main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidator.java

```diff
@@ -53,38 +51,33 @@ public class TripAndShapeDistanceValidator extends FileValidator {
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    List<String> uniqueShapeIds =
-        shapeTable.getEntities().stream()
-            .map(GtfsShape::shapeId)
-            .distinct()
-            .collect(Collectors.toList());
-
-    uniqueShapeIds.forEach(
-        shapeId -> {
-          double maxShapeDist =
-              shapeTable.getEntities().stream()
-                  .filter(s -> s.shapeId().equals(shapeId))
-                  .mapToDouble(GtfsShape::shapeDistTraveled)
-                  .max()
-                  .orElse(Double.NEGATIVE_INFINITY);
-
-          tripTable
-              .byShapeId(shapeId)
-              .forEach(
-                  trip -> {
-                    double maxStopTimeDist =
-                        stopTimeTable.byTripId(trip.tripId()).stream()
-                            .mapToDouble(GtfsStopTime::shapeDistTraveled)
-                            .max()
-                            .orElse(Double.NEGATIVE_INFINITY);
-
-                    if (maxStopTimeDist > maxShapeDist) {
-                      noticeContainer.addValidationNotice(
-                          new TripDistanceExceedsShapeDistanceNotice(
-                              trip.tripId(), shapeId, maxStopTimeDist, maxShapeDist));
-                    }
-                  });
-        });
+    shapeTable
+        .byShapeIdMap()
+        .forEach(
+            (shapeId, shape) -> {
+              double maxShapeDist =
+                  shapeTable.byShapeId(shapeId).stream()
+                      .mapToDouble(GtfsShape::shapeDistTraveled)
+                      .max()
```

Comment:
Seems like the `ShapeIncreasingDistanceValidator` validates that `Values must increase along with shape_pt_sequence` with an `ERROR` severity. Should we always assume that the rule is respected (cc: @davidgamez @emmambd)? It would make the validator a little more efficient.

Reply:
After discussion we concluded that it would be better not to depend on the fact that values increase with the sequence number. 
To be effective we would need to have a dependency between ShapeIncreasingDistanceValidator 
 and TripAndShapeDistanceValidator, meaning don't run the latter if the former is in error. 
Since we don't have this dependency, we cannot guarantee that the values increase.